### PR TITLE
Add enum example using keyof typeof

### PR DESCRIPTION
**Pull request by styfle**
I always forget this syntax.

Please add it to the handbook.

Fixes https://github.com/Microsoft/TypeScript/issues/14106

### CODE REVIEW

**Comment by tjallingt** [comment]
Should probably include a note that this doesn't work for enums with custom values:

```ts
enum LogLevel {
  ERROR = 'error',
  WARN = 'warn',
  INFO = 'info',
  DEBUG = 'debug',
}
```

since the type LogLevelStrings will be `'ERROR' | 'WARN' | 'INFO' | 'DEBUG'` instead of  `'error' | 'warn' | 'info' | 'debug'`

**Comment by styfle** [comment]
@tjallingt It works for string enums too. I reworded the description to be clear that this is getting the keys and not the values.

See [this example](https://typescript-play.js.org/#code/KYOwrgtgBAMg9gcxsAbsANlA3gKCvqAUQCViB5YqAXigHJgAnBuB2gGigHUBBYgOSq0A7gEMGIdlACSfAGJlBASxAAzOJIAihAEIBVAOKCAJsABGYBLRwBfHDgAuATwAOwWImRp0AZXsNlCADO1FAA1sCOcCpQTq5R7kioGADcOCpgIADG9opwIFDO-iD2UhDOLPYixQAU4Y4AXAmeGL5FQRwQwIGBIgjAjYF+AQCU2HgEmXmDUODQNPCJXgDadQC6qQRQitHVs1AAPPMeSegAdDz8o7ibE1Nw6MCn6IjVtAtQD15hEVuBjZJ1YYbG5QSYgQL3R7PBCvd6fDBQFAidBgNyKP50DizIHjG5giEPJ4vN6ID4nKCdbq9NEYySUnp9HGbWy2QrKEplCpVeyvEjkYiSWgAFQAFujflARBSugzgLQgUA) and try running the JS output.

**Comment by tjallingt** [comment]
Great! I find the updated text much clearer :+1:
And yea i meant to say that the way it works is a little bit less clear when concerning enums with custom values. (Yea it says keyof but it doesn't hurt being explicit about what part is actually the key ;p)

File: pages/Enums.md [hunk 1]
```diff
@@ -223,6 +223,32 @@ function f(obj: { X: number }) {
...
+
+Even thought Enums are real objects that exist at runtime, the `keyof` keyword works differently than you might expect for typical objects. Instead, use `keyof typeof` to get a Type that represents all Enum keys as strings.
+
+```ts
+enum LogLevel {
+    ERROR, WARN, INFO, DEBUG
+}
+
+/**
+ * This is equivalen to:
```
Comment (RyanCavanaugh) [inline pages/Enums.md]: equivalent

Comment (styfle) [reply pages/Enums.md]: @RyanCavanaugh Nice! Fixed 🔧

**Comment by orta** [comment]
Cool, thanks!

**Pull request merged**